zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -1139,15 +1150,15 @@ object JsonCodec {
             if (first) first = false
             else {
               out.write(',')
-              if (indent.isDefined) pad(indent_, out)
+              if (doPrettyPrint) pad(indent_, out)
```

Comment:
I think the common case is not pretty printing. Should that not be the first branch?

Reply:
The branch prediction for the common (non pretty printing) case is a smaller issue comparing to overhead of `.isDefined` , `.isEmpty` virtual calls and sub subsequent equality  comparison with `None`.